pkg/workload/histogram/exporter/openmetrics_exporter.go

```diff
@@ -135,18 +137,24 @@ func (o *OpenMetricsExporter) Labels() []*prom.LabelPair {
 	return o.labels
 }
 
-func (o *OpenMetricsExporter) SetLabels(labels *map[string]string) {
+func (o *OpenMetricsExporter) SetLabels(labels *map[string]string) error {
 	var labelValues []*prom.LabelPair
 
 	for label, value := range *labels {
-		labelName := sanitizeOpenMetricsLabels(label)
-		labelValue := value
+		if util.InvalidCharKeyRegex.MatchString(label) {
+			return errors.Errorf("invalid label key %q", label)
```

Comment:
Instead of erroring out, what if we just sanitize the labels/values here? That's what we do in roachprod for adding VM labels and its easier than making the caller need to know what valid sanitization rules are.